### DISCUSSION TITLE
How to restrict gatherer run to only 'some' provider

### DISCUSSION THREAD

**Discussion by angelabriel** [trento-project/wanda#121] [Q&A] [answered]
How to handle the situation that a check/gatherer should only run on one or two of our supported provider but not on the others.
As the object to check (e.g. reading a config file) will be not available on these 'other' providers, the gatherer will fail for these providers and as a result the whole check will fail even that these 'other' providers are not relevant for the expected result of the check.

I tried (searching for a workaround)

```
values:
  - name: expected_SBD_PACEMAKER
    default: notUsed
    conditions:
      - value: yes
        when: env.provider == "azure" || env.provider == "kvm"

expectations:
  - name: expectations_SBD_PACEMAKER
    expect: |
       if values.expected_SBD_PACEMAKER != "notUsed" {
         facts.sbd_SBD_PACEMAKER == values.expected_SBD_PACEMAKER
       } else {
         true
       }
```

But does not help as the gatherer detected the missing file on the system and returned an error, which prevent the evaluation of 'expect:'

```
[{"agent_id":"e2053298-8c2f-5a54-8d6e-0fa749028e32","facts":[{"check_id":"0B6DB2","message":"error reading sbd configuration file: could not open sbd config file: open /etc/sysconfig/sbd: no such file or directory","name":"sbd_SBD_PACEMAKER","type":"sbd-config-file-error"}],"message":"Fact gathering error occurred during the execution","type":"fact_gathering_error"}],"check_id":"0B6DB2","expectation_results":[],"result":"critical"}
```

So what will be the correct way?

**Comment by fabriziosestito** [comment, marked as answer]
@angelabriel a feature to filter checks by the provider is on the radar but not yet implemented. The rationale is that most likely the filtering will be done in the dashboard already, during the check selection, so the user will not be able to run a check for another provider.

I suggest not trying to workaround the expectation language in this case, but just assume that the check will only run on certain providers. 

If you wish, you can add a comment somewhere in the yaml so we can trace which checks must be filtered when we implement the filtering in the next iteration.

**Reply by angelabriel** [reply]
ok, thank for the explanation. Will add a comment to the checks.
